internal/cluster/validator.go

```diff
@@ -213,48 +238,57 @@ func isVipDhcpAllocationAndOVN(cluster *common.Cluster) bool {
 	return isVipDhcpAllocation && isNetworkTypeOVN
 }
 
-func (v *clusterValidator) isIngressVipDefined(c *clusterPreprocessContext) (ValidationStatus, string) {
+func (v *clusterValidator) areIngressVipsDefined(c *clusterPreprocessContext) (ValidationStatus, string) {
 	if swag.BoolValue(c.cluster.UserManagedNetworking) {
-		return ValidationSuccess, "The Ingress virtual IP is not required: User Managed Networking"
+		return ValidationSuccess, "Ingress virtual IPs are not required: User Managed Networking"
 	}
-	machineCidrDefined, _ := v.isMachineCidrDefined(c)
-	if swag.BoolValue(c.cluster.VipDhcpAllocation) && !validationStatusToBool(machineCidrDefined) {
-		return ValidationPending, "The Machine Network CIDR is undefined"
+	if len(c.cluster.IngressVips) > 0 {
+		return ValidationSuccess, "Ingress virtual IPs are defined."
 	}
-
-	if c.cluster.IngressVip != "" {
-		return ValidationSuccess, "The Ingress virtual IP is defined."
-	}
-
+	machineCidrDefined, _ := v.isMachineCidrDefined(c)
 	if swag.BoolValue(c.cluster.VipDhcpAllocation) {
+		if !validationStatusToBool(machineCidrDefined) {
+			return ValidationPending, "The Machine Network CIDR is undefined"
+		}
```

Comment:
Before this PR the order of validations is
1) check if machine cidr is defined
2) check if ingress vips are defined

After this PR, the order changes so
* we check if ingress VIPs are defined and return Success if yes
* only after this, we check if machine cidr is defined

So I can imagine a scenario in which you have Ingress VIP but you are missing Machine Networks. Previously this would fail the validation. With this change, the validation will pass. Am I right? If that's the case, I'm not sure it's desired behaviour.

On the other hand in the old `areApiVipsDefined` function the order was always reversed, so I'm not sure the original author ever thought about this...

Reply:
I changed it back to the original order